tests/testsuite/cargo_information/specify_version_with_url_but_registry_is_not_matched/mod.rs

```diff
@@ -0,0 +1,23 @@
+use cargo_test_macro::cargo_test;
+use cargo_test_support::{curr_dir, registry::RegistryBuilder};
+
+use super::cargo_info;
+
+#[cargo_test]
+fn case() {
+    let _ = RegistryBuilder::new()
+        .alternative()
+        .no_configure_token()
+        .build();
+    cargo_test_support::registry::Package::new("my-package", "99999.0.0-alpha.1+my-package")
+        .alternative(true)
+        .publish();
+
+    cargo_info()
+        .arg("https://crates.io/my-package")
```

Comment:
I think this is a bug. Because the passed registry is an alternative registry, we should bail out an error for it.

Reply:
Maybe something to note in the issue tracker / proposal for merge